doc/enterprise/auth/auth.md

```diff
@@ -0,0 +1,234 @@
+# Understanding Pachyderm access controls
+
+If access controls are activated, each data repository, or repo,
+in Pachyderm has an Access Control List (ACL) associated with it.
+The ACL includes:
+
+- `READERs` - users who can read the data versioned in the repo.
+- `WRITERs` - users with `READER` access who can also submit
+additions, deletions, or modifications of data into the repo.
+- `OWNERs` - users with READER and WRITER access who can also
+modify the repo's ACL.
+
+Pachyderm defines the following user account types:
+
+* **GitHub user** is a user account that is associated with
+a GitHub account and logs in through the GitHub OAuth flow. If you do not
+use any third-party identity provider, you use this option. When a user tries
+to log in with a GitHub account, the system verifies the identity and
+sends a Pachyderm token for that account.
+* **Robot user** is a user account that logs by using a pach-generated authentication
+token. Typically, you create a user in simplified workflow scenarios, such
+as initial SAML configuration.
+* **Pipeline** is a user account that Pachyderm creates for your
+data pipelines. Pipelines inherit access control from its creator.
+* **SAML user** is a user account that is associated with a Security Assertion
+Markup Language (SAML) identity provider, such as Okta.
+When a user tries to log in through a SAML ID provider, the system
+confirms the identity, associates
+that identity with a SAML identity provider account, and responds with
+the SAML identity provider token for that user.
```

Comment:
I think this is almost perfect, the one disambiguation I'd suggest is that "the SAML identity provider token" could refer to the token that the SAML identity provider sends to pachyderm saying "I, Okta, promise that this is Svetlana", which Pachyderm doesn't send back (it verifies it and then throws it away). Instead, Pachyderm creates a new token that's meaningful to Pachyderm saying "this is the user Svetlana in the Okta system that I know about" (essentially pachyderm proving to itself that you're Svetlana, rather than Okta proving to Pachyderm that you're Svetlana)